
# robot kinematics
float32 hand_angle
float32[] hand_pose
float32[] hand_sliding

# object kinematics
float32 object_angle 
float32[] object_pose
float32[] object_sliding 
float32[] pivot_locations

# ground 
float32 ground_angle

# perception 
string shape_name
int8 apriltag_id
bool object_dectected